tests/subsys/net/openthread/rpc/client/commissioning/src/main.c

```diff
@@ -0,0 +1,191 @@
+}
+
+/* Test serialization of otDatasetIsCommissioned() */
+ZTEST(ot_rpc_commissioning_cli, test_otDatasetIsCommissioned)
+{
+	bool result;
+
+	mock_nrf_rpc_tr_expect_add(RPC_CMD(OT_RPC_CMD_DATASET_IS_COMMISSIONED), RPC_RSP(CBOR_TRUE));
+	result = otDatasetIsCommissioned(NULL);
+	mock_nrf_rpc_tr_expect_done();
+	zassert_true(result);
+
+	mock_nrf_rpc_tr_expect_add(RPC_CMD(OT_RPC_CMD_DATASET_IS_COMMISSIONED),
+				   RPC_RSP(CBOR_FALSE));
+	result = otDatasetIsCommissioned(NULL);
+	mock_nrf_rpc_tr_expect_done();
+	zassert_false(result);
+}
+
+/* Test serialization of otDatasetSetActiveTlvs() */
+ZTEST(ot_rpc_commissioning_cli, test_otDatasetSetActiveTlvs)
+{
+	otError error;
+
+	const otOperationalDatasetTlvs dataset = {
+		{LISTIFY(OT_OPERATIONAL_DATASET_MAX_LENGTH, LSFY_IDENTITY, (,))},
+		OT_OPERATIONAL_DATASET_MAX_LENGTH};
+
+	mock_nrf_rpc_tr_expect_add(RPC_CMD(OT_RPC_CMD_DATA_SET_ACTIVE_TLVS, TLVS),
+				   RPC_RSP(OT_ERROR_NONE));
+	error = otDatasetSetActiveTlvs(NULL, &dataset);
+	mock_nrf_rpc_tr_expect_done();
+	zassert_equal(error, OT_ERROR_NONE);
+}
+
+/* Test incoming parameters validation of otDatasetSetActiveTlvs() */
+ZTEST(ot_rpc_commissioning_cli, test_otDatasetSetActiveTlvs_negative)
+{
+	otError error;
+	const otOperationalDatasetTlvs dataset = {{0}, OT_OPERATIONAL_DATASET_MAX_LENGTH + 1};
+
+	error = otDatasetSetActiveTlvs(NULL, &dataset);
+	zassert_equal(error, OT_ERROR_INVALID_ARGS);
+	error = otDatasetSetActiveTlvs(NULL, NULL);
+	zassert_equal(error, OT_ERROR_INVALID_ARGS);
+}
+
+/* Test serialization of otDatasetGetActiveTlvs() */
+ZTEST(ot_rpc_commissioning_cli, test_otDatasetGetActiveTlvs)
+{
+	otError error;
+	otOperationalDatasetTlvs dataset;
+	const otOperationalDatasetTlvs expected_dataset = {
+		{LISTIFY(OT_OPERATIONAL_DATASET_MAX_LENGTH, LSFY_IDENTITY, (,))},
+		OT_OPERATIONAL_DATASET_MAX_LENGTH};
+
+	mock_nrf_rpc_tr_expect_add(RPC_CMD(OT_RPC_CMD_DATA_GET_ACTIVE_TLVS), RPC_RSP(TLVS));
+	error = otDatasetGetActiveTlvs(NULL, &dataset);
+	mock_nrf_rpc_tr_expect_done();
+	zassert_equal(error, OT_ERROR_NONE);
+	zassert_mem_equal(&dataset, &expected_dataset, sizeof(otOperationalDatasetTlvs));
+}
+
+/* Test NULL replay on otDatasetGetActiveTlvs() and parameter validation. */
+ZTEST(ot_rpc_commissioning_cli, test_otDatasetGetActiveTlvs_null)
+{
+	otError error;
+	otOperationalDatasetTlvs dataset;
+
+	mock_nrf_rpc_tr_expect_add(RPC_CMD(OT_RPC_CMD_DATA_GET_ACTIVE_TLVS), RPC_RSP(CBOR_NULL));
+	error = otDatasetGetActiveTlvs(NULL, &dataset);
+	mock_nrf_rpc_tr_expect_done();
+	zassert_equal(error, OT_ERROR_NOT_FOUND);
+
+	error = otDatasetGetActiveTlvs(NULL, NULL);
+	zassert_equal(error, OT_ERROR_NOT_FOUND);
+}
+
+static void discover_cb(otActiveScanResult *result, void *ctx)
+{
+	uint8_t ext_addr[] = {LISTIFY(OT_EXT_ADDRESS_SIZE, LSFY_IDENTITY, (,))};
+	uint8_t nwk_name[] = {LISTIFY(OT_NETWORK_NAME_MAX_SIZE, LSFY_IDENTITY, (,)), 0};
+	uint8_t ext_pan_id[] = {LISTIFY(OT_EXT_PAN_ID_SIZE, LSFY_IDENTITY, (,)), 0};
```

Comment:
ext_pan_id an steer_data don't have to be null-terminated

Reply:
Sure, leftover after debugging. I'll remove it.